### PR TITLE
[PRM-1507] Removed old API data and feature switch, along with changing LocalDateTime to LocalDate

### PR DESCRIPTION
**Pull request by JBlacklaws**

### CODE REVIEW

File: app/models/AppealData.scala [hunk 1]
```diff
@@ -16,20 +16,19 @@
...
-                     ) extends AppealInformation[A]
+                       startDate: A,
+                       endDate: A,
+                       dueDate: A,
+                       dateCommunicationSent: A
+                     )
 
 object AppealData {
-  implicit val format: Format[AppealData[LocalDateTime]] = Json.format[AppealData[LocalDateTime]]
+  implicit val format: Format[AppealData[LocalDate]] = Json.format[AppealData[LocalDate]]
```
Comment (LewisMitchell) [inline app/models/AppealData.scala:32]: We can revert this back to how it was before, by removing the type parameter ([A]) from the case class and changing startDate: A to startDate: LocalDate etc.

File: app/models/appeals/AppealSubmission.scala [hunk 2]
```diff
@@ -437,7 +437,7 @@ case class AppealSubmission(
                              sourceSystem: String,
                              taxRegime: String,
                              customerReferenceNo: String,
-                             dateOfAppeal: LocalDateTime,
+                             dateOfAppeal: LocalDate,
```
Comment (LewisMitchell) [inline app/models/appeals/AppealSubmission.scala:440]: Checking the API, the dateOfAppeal is a LocalDateTime

File: app/models/upload/UploadDetails.scala [hunk 3]
```diff
@@ -18,12 +18,12 @@ package models.upload
...
 import play.api.libs.json.{Json, OFormat}
 
-import java.time.LocalDateTime
+import java.time.LocalDate
 
 case class UploadDetails(
                           fileName: String,
                           fileMimeType: String,
-                          uploadTimestamp: LocalDateTime,
+                          uploadTimestamp: LocalDate,
```
Comment (LewisMitchell) [inline app/models/upload/UploadDetails.scala:26]: We get this from Upscan, this isn't to do with the feature switch

File: app/models/upload/UploadJourney.scala [hunk 4]
```diff
@@ -18,15 +18,15 @@ package models.upload
...
+import java.time.LocalDate
 
 case class UploadJourney (
                            reference: String,
                            fileStatus: UploadStatusEnum.Value,
                            downloadUrl: Option[String] = None,
                            uploadDetails: Option[UploadDetails] = None,
                            failureDetails: Option[FailureDetails] = None,
-                           lastUpdated: LocalDateTime = LocalDateTime.now(),
+                           lastUpdated: LocalDate = LocalDate.now(),
```
Comment (LewisMitchell) [inline app/models/upload/UploadJourney.scala:29]: This is not to do with the feature switch, keep it as LocalDateTime

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, the file upload changes need to be reverted

File: app/models/upload/UploadDetails.scala [hunk 5]
```diff
@@ -17,13 +17,12 @@
 package models.upload
 
 import play.api.libs.json.{Json, OFormat}
-
-import java.time.LocalDate
```
Comment (LewisMitchell) [inline app/models/upload/UploadDetails.scala]: You can remove the LocalDate import now

File: test/controllers/internal/UpscanCallbackControllerSpec.scala [hunk 6]
```diff
@@ -24,8 +24,8 @@ import play.api.libs.json.{JsValue, Json}
 import play.api.test.Helpers._
 import repositories.UploadJourneyRepository
 import uk.gov.hmrc.mongo.cache.DataKey
+import java.time.{LocalDate, LocalDateTime}
```
Comment (LewisMitchell) [inline test/controllers/internal/UpscanCallbackControllerSpec.scala:27]: You can remove the LocalDate import now

File: test/helpers/SessionAnswersHelperSpec.scala [hunk 7]
```diff
@@ -25,8 +25,8 @@ import play.api.mvc.AnyContent
 import play.api.test.Helpers._
 import repositories.UploadJourneyRepository
 import utils.SessionKeys
+import java.time.{LocalDate, LocalDateTime}
```
Comment (LewisMitchell) [inline test/helpers/SessionAnswersHelperSpec.scala:28]: You can remove the LocalDate import now

File: test/services/AppealServiceSpec.scala [hunk 8]
```diff
@@ -30,7 +30,7 @@ import play.api.test.Helpers._
 import services.monitoring.JsonAuditModel
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 import utils.{SessionKeys, UUIDGenerator}
-import java.time.LocalDate
+import java.time.{LocalDate, LocalDateTime}
```
Comment (LewisMitchell) [inline test/services/AppealServiceSpec.scala:33]: You can remove the LocalDate import now

**Review by LewisMitchell — DISMISSED** [review]

**Review by AndrewRButcher — APPROVED** [review]

**Pull request merged**